linera-core/src/chain_worker/state/mod.rs

```diff
@@ -423,6 +424,18 @@ where
     async fn create_network_actions(&self) -> Result<NetworkActions, WorkerError> {
         let mut heights_by_recipient = BTreeMap::<_, BTreeMap<_, _>>::new();
         let mut targets = self.chain.outboxes.indices().await?;
+
+        if let Some(limit) = self.config.network_actions_batch_limit {
+            if limit > targets.len() / 2 {
+                let elements_to_discard = targets.len().saturating_sub(limit);
+                targets.partial_shuffle(&mut thread_rng(), elements_to_discard);
+                targets.drain(0..elements_to_discard);
+            } else {
+                targets.partial_shuffle(&mut thread_rng(), limit);
+                targets.truncate(limit);
+            }
+        }
+
         if let Some(tracked_chains) = self.tracked_chains.as_ref() {
             let publishers = self
                 .chain
```

Comment:
Personally, I was only worried about returning too many `NetworkAction`s, so I'm a little bit on fence regarding the randomization idea. It could also be counter-productive.